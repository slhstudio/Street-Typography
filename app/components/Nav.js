import React from 'react';
import { NavLink } from 'react-router-dom';

const Nav = (props) => {
    return (
      <ul className='nav'>
        <li>
          <NavLink exact activeClassName='active' to='/'>
            Explore Photos
          </NavLink>
        </li>

        <li>
          <NavLink activeClassName='active' to='/add'>
            Add Photos
          </NavLink>
        </li>

        <li>
          <NavLink activeClassName='active' to='/mine'>
            My Photos
          </NavLink>
        </li>
        <li className='greeting'>Hi, {props.name}!</li>
        <li>
          <NavLink activeClassName='active' to='/logIn'>
            {props.logged}
          </NavLink>
        </li>
      </ul>
    )
  }


export default Nav