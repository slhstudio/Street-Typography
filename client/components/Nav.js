import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const NavList = styled.ul.attrs({
  className: 'navlist'
})`
  display: flex;
  padding-top: 1em;
  margin: 0 3em 0 3em;
  
  .active {
    font-weight: bold;
  }
  li {
    margin: 0 1em 0 1em;
  }
  a {
    color: darkcyan;
  }
  li:last-child {
    margin-left: auto
    a {
      color: white;
      margin-left: 2em;
    }
  }
`;

const Greeting = styled.div.attrs({
  className: 'greeting'
})`
  color: darkcyan;
  display: flex;
  flex-direction: row;
`;

const Nav = (props) => {
    return (
      <NavList>
        <li>
          <NavLink exact activeClassName='active' to='/'>
            Explore Photos
          </NavLink>
        </li>

        <li>
          <NavLink activeClassName='active' to='/add'>
            Add Photos
          </NavLink>
        </li>

        <li>
          <NavLink activeClassName='active' to='/mine'>
            My Photos
          </NavLink>
        </li>
       
        <li>
          { !props.isUser
          ? <NavLink activeClassName='active' to='/logIn'>
              {props.logged}
            </NavLink>
          : <Greeting {...props}>
              Hi, {props.name}!
              <a href='/auth/logout'>
                {props.logged}
              </a>
            </Greeting>
          }
        </li>
      </NavList>
    )
  }

  Nav.propTypes = {
    logged: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    isUser: PropTypes.bool.isRequired,
  }

export default Nav