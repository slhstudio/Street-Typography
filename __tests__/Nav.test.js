import React from 'react';
import Nav from '../client/components/Nav';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const logged1 = 'Log In';
const logged2 = 'Log Out';
const name = 'Sarah';

describe('<Nav/>', () => {
  it ('renders and displays properly if there is no a user', () => {
    const wrapper = shallow(<Nav isUser={false} logged={logged1}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
   });
   it ('renders and displays properly if there IS a user', () => {
    const wrapper = shallow(<Nav isUser={true} logged={logged2} name={name}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
   });
});