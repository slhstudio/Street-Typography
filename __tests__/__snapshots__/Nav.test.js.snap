// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> renders and displays properly if there IS a user 1`] = `
<ul
  className="nav"
>
  <li>
    <NavLink
      activeClassName="active"
      aria-current="page"
      exact={true}
      to="/"
    >
      Explore Photos
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/add"
    >
      Add Photos
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/mine"
    >
      My Photos
    </NavLink>
  </li>
  <li>
    <div
      className="row greeting"
    >
      Hi, 
      Sarah
      !
      <a
        href="/auth/logout"
      >
        Log Out
      </a>
    </div>
  </li>
</ul>
`;

exports[`<Nav/> renders and displays properly if there is no a user 1`] = `
<ul
  className="nav"
>
  <li>
    <NavLink
      activeClassName="active"
      aria-current="page"
      exact={true}
      to="/"
    >
      Explore Photos
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/add"
    >
      Add Photos
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/mine"
    >
      My Photos
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/logIn"
    >
      Log In
    </NavLink>
  </li>
</ul>
`;
